glass/src/lib/native/cpp/other/Field2D.cpp

```diff
@@ -952,15 +952,12 @@ void glass::DisplayField2DSettings(Field2DModel* model) {
       return;
     }
     PushID(name);
-    auto& objRef = field->m_objects[name];
-    if (!objRef) {
-      objRef = std::make_unique<ObjectInfo>(GetStorage());
-    }
-    auto obj = objRef.get();
 
     wpi::SmallString<64> nameBuf{name};
     if (ImGui::CollapsingHeader(nameBuf.c_str())) {
-      obj->DisplaySettings();
+      auto& obj =
+          field->m_objects.try_emplace(name, GetStorage()).first->second;
+      obj.DisplaySettings();
```

Comment:
I'll note that this is a behavior change (`ObjectInfo` won't be created and added if `ImGui::CollapsingHeading()` returns false), but I don't know enough about the context to know if that matters or not.

Reply:
Yes, it was intentional on my part.  We have the same logic in the display side so this should be fine.